phoenix-core/src/main/java/org/apache/phoenix/iterate/BaseResultIterators.java

```diff
@@ -491,6 +476,22 @@ private static void optimizeProjection(StatementContext context, Scan scan, PTab
                 }
             }
         }
+        for (Entry<byte[], NavigableSet<byte[]>> entry : familyMap.entrySet()) {
```

Comment:
This is one of the bug fixes. Above we add columns and column families sometimes that we would miss here in the trackedColumnBitset as well as the column tracker.

This showed in queries like ... WHERE (a,b) IN ((..., ...), (..., ...)) when 'a' is indexed by a local index and 'b' is included. I expected there are potential other scenarios where this is wrong.